### PR TITLE
Adding configs related to DCLM

### PR DESCRIPTION
**Pull request by abhinavg4**
DCLM 7B related configs

### CODE REVIEW

**Comment by Ivan-Zhou** [comment]
LGTM. Can you run pre-commit checks to fix the pre-commit issue? I think it is just some formatting issue. 

```
pre-commit run --all-files
```

File: src/levanter/models/llama.py [hunk 1]
```diff
@@ -64,6 +65,7 @@ class LlamaConfig(HFCompatConfig):
     activation_function: str = "silu"
     initializer_range: float = 0.02
     layer_norm_epsilon: float = 1e-5
+    z_loss_weight: float = 0.0
```
Comment (dlwh) [inline src/levanter/models/llama.py:68]: i would rather this not be a property of the model's config but an option on TrainLmConfig, and define a loss function in train_lm and pass it into trainer

File: src/levanter/models/lm_model.py [hunk 2]
```diff
@@ -131,9 +131,12 @@ def compute_loss(
         logits = logits.astype(jnp.float32)
         targets = hax.roll(example.tokens, -1, axis=self.Pos.name)
         target_y = hax.nn.one_hot(targets, self.Vocab, dtype=logits.dtype)
-        loss = cross_entropy_loss(
-            logits, self.Vocab, target_y, reduction, reduction_axis=reduction_axis, where=example.loss_mask
-        )
+        if hasattr(self.config, "z_loss_weight") and self.config.z_loss_weight > 0:
```
Comment (dlwh) [inline src/levanter/models/lm_model.py:134]: really don't like using this here. much cleaner to just pull out the loss function

File: src/levanter/main/train_lm.py [hunk 3]
```diff
@@ -19,12 +19,29 @@
...
 from levanter.optim import AdamConfig, OptimizerConfig
 from levanter.trainer import Trainer, TrainerConfig
+from levanter.types import ComputeLossFunction, M, X
 from levanter.utils.jax_utils import parameter_count
 
 
 logger = logging.getLogger(__name__)
 
 
+class ModuleComputeZLoss(ComputeLossFunction[M, X]):
```
Comment (dlwh) [inline src/levanter/main/train_lm.py:29]: i still don't like this but I think I can't really articulate what I want. i'm gonna push a change to my fork
